### PR TITLE
[zh-cn]: add the translation of ArrayBuffer.transfer()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#16958

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+transfer(newByteLength)
+```
+
+### 参数
+
+- `newByteLength` {{optional_inline}}
+  - : 新的 `ArrayBuffer` 的 {{jsxref("ArrayBuffer/byteLength", "byteLength")}}。默认为此 `ArrayBuffer` 的 `byteLength`。
+    - 如果 `newByteLength` 小于此 `ArrayBuffer` 的 `byteLength`，“溢出”的字节将被丢弃。
+    - 如果 `newByteLength` 大于此 `ArrayBuffer` 的 `byteLength`，剩下的的字节将用零填充。
+    - 如果此 `ArrayBuffer` 是可调整大小的，`newByteLength` 一定不能大于其 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:25]: 感觉这样会好一些，上面那一处也可以考虑一下

```suggestion
  - : 新的 `ArrayBuffer` 的 {{jsxref("ArrayBuffer/byteLength", "byteLength")}}。默认为当前 `ArrayBuffer` 的 `byteLength`。
    - 如果 `newByteLength` 小于 `ArrayBuffer` 的 `byteLength`，“溢出”的字节将被丢弃。
    - 如果 `newByteLength` 大于 `ArrayBuffer` 的 `byteLength`，剩下的的字节将用零填充。
    - 如果 `ArrayBuffer` 是可调整大小的，`newByteLength` 一定不能大于其 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+---
+title: ArrayBuffer.prototype.transfer()
+slug: Web/JavaScript/Reference/Global_Objects/ArrayBuffer/transfer
+l10n:
+  sourceCommit: fb236a32b20d1985b35b1c0c25ce99f2a0e0d5fe
+---
+
+{{JSRef}} {{SeeCompatTable}}
+
+{{jsxref("ArrayBuffer")}} 实例的 **`transfer()`** 方法创建一个内容与此缓冲区相同的新 `ArrayBuffer` 实例，然后将此缓冲区分离。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:10]:
```suggestion
{{jsxref("ArrayBuffer")}} 实例的 **`transfer()`** 方法创建一个内容与该缓冲区相同的新 `ArrayBuffer` 实例，然后将源缓冲区分离。
```

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md (hunk 1 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:25]:
```suggestion
  - : 新的 `ArrayBuffer` 的 {{jsxref("ArrayBuffer/byteLength", "byteLength")}}。默认为当前 `ArrayBuffer` 的 `byteLength`。
    - 如果 `newByteLength` 小于当前 `ArrayBuffer` 的 `byteLength`，“溢出”的字节将被丢弃。
    - 如果 `newByteLength` 大于当前 `ArrayBuffer` 的 `byteLength`，剩下的的字节将用零填充。
    - 如果当前的 `ArrayBuffer` 是可调整大小的，`newByteLength` 一定不能大于其 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+### 参数
+
+- `newByteLength` {{optional_inline}}
+  - : 新的 `ArrayBuffer` 的 {{jsxref("ArrayBuffer/byteLength", "byteLength")}}。默认为当前 `ArrayBuffer` 的 `byteLength`。
+    - 如果 `newByteLength` 小于当前 `ArrayBuffer` 的 `byteLength`，“溢出”的字节将被丢弃。
+    - 如果 `newByteLength` 大于当前 `ArrayBuffer` 的 `byteLength`，剩下的的字节将用零填充。
+    - 如果当前的 `ArrayBuffer` 是可调整大小的，`newByteLength` 一定不能大于其 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}}。
+
+### 返回值
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:27]: 后面的部分也修改一下

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+一个新的 {{jsxref("ArrayBuffer")}} 对象。其内容被初始化为此 `ArrayBuffer` 的内容，如果有额外的字节。则填充为零。当且仅当此 `ArrayBuffer` 是可调整大小的，新 `ArrayBuffer` 才是可调整大小的，在这种情况下，其 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}} 和此 `ArrayBuffer` 的相同。原始 `ArrayBuffer` 将被分离。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:29]:
```suggestion
一个新的 {{jsxref("ArrayBuffer")}} 对象。其内容被初始化为当前 `ArrayBuffer` 的内容，如果有额外的字节。则填充为零。当且仅当当前 `ArrayBuffer` 是可调整大小的，新 `ArrayBuffer` 才是可调整大小的，在这种情况下，其 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}} 和当前 `ArrayBuffer` 的相同。当前 `ArrayBuffer`（在调用后）将被分离。
```

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+### 异常
+
+- {{jsxref("RangeError")}}
+  - : 如果此 `ArrayBuffer` 是可调整大小的并且 `newByteLength` 大于此 `ArrayBuffer` 的 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}}，则抛出此错误。
+- {{jsxref("TypeError")}}
+  - : 如果此 `ArrayBuffer` 已经分离，则抛出此错误。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:36]: ### 
```suggestion
- {{jsxref("RangeError")}}
  - : 如果当前 `ArrayBuffer` 是可调整大小的并且 `newByteLength` 大于当前 `ArrayBuffer` 的 {{jsxref("ArrayBuffer/maxByteLength", "maxByteLength")}}，则抛出此错误。
- {{jsxref("TypeError")}}
  - : 如果当前 `ArrayBuffer` 已经分离，则抛出此错误。
```

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+## 描述
+
+`transfer()` 方法执行与[结构化克隆算法](/zh-CN/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)相同的操作。它将此 `ArrayBuffer` 的字节复制到一个新的 `ArrayBuffer` 对象中，然后分离此 `ArrayBuffer` 对象。有关更多信息，请参阅[传输 ArrayBuffers](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer#传输_arraybuffer)。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:40]:
```suggestion
`transfer()` 方法执行与[结构化克隆算法](/zh-CN/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)相同的操作。它将当前 `ArrayBuffer` 的字节复制到一个新的 `ArrayBuffer` 对象中，然后分离当前 `ArrayBuffer` 对象。有关更多信息，请参阅[传输 ArrayBuffers](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer#传输_arraybuffer)。
```

File: files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+`transfer()` 保留了此 `ArrayBuffer` 的大小可调整性。如果你希望新的 `ArrayBuffer` 不可调整大小，请使用 {{jsxref("ArrayBuffer/transferToFixedLength", "transferToFixedLength()")}} 代替。没有办法通过传输使长度固定的缓冲区变为可调整大小的缓冲区。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/arraybuffer/transfer/index.md:42]:
```suggestion
`transfer()` 保留了当前 `ArrayBuffer` 的大小可调整性。如果你希望新的 `ArrayBuffer` 不可调整大小，请使用 {{jsxref("ArrayBuffer/transferToFixedLength", "transferToFixedLength()")}} 代替。没有办法通过传输使长度固定的缓冲区变为可调整大小的缓冲区。
```

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**